src/Plugin/CKEditorPlugin/OembedEntities.php

```diff
@@ -64,7 +74,66 @@ public function getConfig(Editor $editor): array {
       'current_route' => $this->currentRouteMatch->getRouteName(),
       'current_route_parameters' => $this->currentRouteMatch->getRawParameters()->all(),
       'Oembed_buttons' => $this->getButtons(),
+      'Oembed_default_buttons' => $this->getDefaultButtons($editor),
     ];
   }
 
+  /**
+   * Returns the default oembed buttons for the editor.
+   *
+   * For each entity type, we determine a button to be used as default so that
+   * we can edit existing embeds where we do not have information about the
+   * original button used to embed them.
+   *
+   * This also takes into account if the button is in fact enabled on the
+   * editor.
+   *
+   * @param \Drupal\editor\Entity\Editor $editor
+   *   The editor.
+   *
+   * @return array
+   *   The default buttons.
+   */
+  protected function getDefaultButtons(Editor $editor): array {
+    $buttons = $this->getButtons();
+    $default_buttons = [];
+    foreach ($buttons as $id => $info) {
+      /** @var \Drupal\embed\EmbedButtonInterface $button */
+      $button = $this->entityTypeManager->getStorage('embed_button')->load($id);
+      $entity_type = $button->getTypeSetting('entity_type');
+      if ($this->isButtonEnabled($id, $editor) && !isset($default_buttons[$entity_type])) {
+        $default_buttons[$entity_type] = $id;
+      }
+    }
+
+    return $default_buttons;
+  }
+
+  /**
+   * Checks if a given button is enabled on the editor.
+   *
+   * @param string $button
+   *   The button ID.
+   * @param \Drupal\editor\Entity\Editor $editor
+   *   The editor.
+   *
+   * @return bool
+   *   Whether the button is enabled in the editor.
+   */
+  protected function isButtonEnabled(string $button, Editor $editor): bool {
+    $enabled = FALSE;
+    $settings = $editor->getSettings();
+    foreach ($settings['toolbar']['rows'] as $row) {
+      foreach ($row as $group) {
+        foreach ($group['items'] as $group_button) {
+          if ($group_button === $button) {
+            $enabled = TRUE;
```

Comment:
Can't we just return `TRUE` here instead of continuing in the loops and return `FALSE` outside the `foreach` statements? This way the `$enabled` wouldn't be needed anymore.